snapshot/snapshotfs/upload.go

```diff
@@ -606,6 +614,12 @@ func uniqueDirectories(dirs []fs.Directory) []fs.Directory {
 	return result
 }
 
+// DirReadError distinguishes an error thrown when attempting to
+// read a directory
+type DirReadError struct {
```

Comment:
Could this be changed to a type redefinition and still have the type assertion above work unmodified?
```suggestion
type DirReadError error
```

Reply:
Tried that first, won't work though. Any error would be fine to be type asserted as a DirReadError.

https://play.golang.org/p/2iQD3e_Ewgd

Reply:
Oh! I see. Interesting.